### PR TITLE
Verify modules will compile for amd64

### PR DESCRIPTION
**Pull request by fredlawl**
The user-mode linux architecture compile hides errors we may encounter when actually compiling for release. Therefore, we need to ensure our modules can at least be compiled against amd64.

Introduce 3 new steps:

1. make allnoconfig 
Ensures that we're not introducing anything that would accidentally break a build where nothing is set.

2. make defconfig
This is our control build.

3. make defconfig + config-integration 
This tests that we can do our control + custom options. i.e. enable modules we want to compile.

These three tests should be enough to satisfy Intel's testing-bot. We'll skip allyesconfig since that takes a very long time to compile. Lastly, enable "continue-on-error" so that we can see the results for all 3 builds at once and not individually.

### CODE REVIEW

File: .github/workflows/crypto-test-harness.yml [hunk 1]
```diff
@@ -43,3 +43,40 @@ jobs:
...
           name: tcrypt_job
           path: ./tcrypt.out
+  cfgcheck:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y bc libelf-dev pahole
+          mkdir -p out
```
Comment (fredlawl) [inline .github/workflows/crypto-test-harness.yml]: Hmm. Does this need to be AFTER the actions/checkout@v3 step? All of the steps failed because of:

```
/home/runner/work/_temp/1f94bbb9-a7e7-4490-a97b-3938df7b1511.sh: line 6: out/config-integration.out: No such file or directory
Error: Process completed with exit code 1.
```

File: .github/workflows/crypto-test-harness.yml [hunk 2]
```diff
@@ -43,3 +43,40 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (fredlawl) [inline .github/workflows/crypto-test-harness.yml]: I think we're going to need to also update these to 4.2.0?
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses

In the build output, we have some notifications about node-16 EOL. So changing these will be their own commit ofc.

File: .github/workflows/crypto-test-harness.yml [hunk 3]
```diff
@@ -43,3 +43,40 @@ jobs:
...
+          make -j$(nproc) > results/config-defconfig.out 2>&1
+      - name: defconfig + config-integration
+        continue-on-error: true
+        run: |
+          cp zeta/test-artifacts/config-integration .
+          make mrproper
+          make defconfig
+          ./scripts/kconfig/merge_config.sh -m .config config-integration
+          make olddefconfig
+          make -j$(nproc) > results/config-integration.out 2>&1
```
Comment (fredlawl) [inline .github/workflows/crypto-test-harness.yml:76]: In my testing container, I couldn't quite get the `make -j$(nproc) > file 2>&1` to work as I wanted it to. I think for now we assume that it is working, because there's no build errors to report anyway for this PR. The artifacts also contain the full build output anyway.

**Review by jopelima — APPROVED** [review]

**Review by armfazh — APPROVED** [review]
